foyer/plugins.py

```diff
@@ -0,0 +1,39 @@
+import foyer
+import glob
+
+def collect_plugins(plugin_names=None):
+    """
+    Detects which plugins are installed
+
+    Arguments
+    ---------
+    plugin_names: list, default=None
+        A list of non-standard plugin names (strings) to additionally
+        check for.
+    """
+
+    plugin_funcs = [func for func in dir(foyer.forcefields)
+                    if "load" in func and "__" not in func]
+
+    if plugin_names:
+        plugin_funcs += plugin_names
+
+    plugins = dict()
+    for plugin_func in plugin_funcs:
+        plugin_loader_func = eval("foyer.forcefields.{}".format(plugin_func))
+        # Assumes all plugins are named "load_{plugin_name}"
+        plugin_name = "_".join(plugin_func.split("_")[1:])
+        # TODO: plugin_version = get_version_info
```

Comment:
Any initial thoughts on how we should store this information? I see we want it to be accessible from the loader but I'm not sure how to go about doing that - maybe these functions should be expanded into (lightweight) classes?

Reply:
Check out my [plugins branch for the PFE repo](https://github.com/uppittu11/forcefield_perfluoroethers/tree/plugins). To demonstrate how FF versioning would work, I added some dummy forcefield versions and file organization for adding forcefields versions. Version and xml_path are returned when the forcefield is loaded, and can be easily accessed by the plugin detector in foyer:

```python3
>>> import foyer
>>> pfe = foyer.forcefields.load_PFE()
>>> pfe.version
'1.0.0'
>>> pfe.xml_path
['/Users/parashara/Documents/devel/git/forcefield_perfluoroethers/perfluoroethers/xml/current/PFE.xml']
>>> pfe = foyer.forcefields.load_PFE("0.0.2")
>>> pfe.version
'0.0.2'
>>> pfe.xml_path
['/Users/parashara/Documents/devel/git/forcefield_perfluoroethers/perfluoroethers/xml/versions/0.0.2/PFE.xml']
```